core/GlobalState.h

```diff
@@ -264,6 +283,9 @@ class GlobalState final {
     // If lastCommittedLSPEpoch != currentlyProcessingLSPEpoch, then GlobalState is currently running a slow path
     // containing edits (lastCommittedLSPEpoch, currentlyProcessingLSPEpoch].
     const std::shared_ptr<std::atomic<u4>> lastCommittedLSPEpoch;
+    // Task to run when pre-empting typechecking. Outer shared_ptr is shared amongst all GlobalState instances that
+    // share a common lineage, whereas contents of inner shared_ptr is atomically replaced during preemption.
+    const std::shared_ptr<std::shared_ptr<lsp::Task>> preemptTask;
```

Comment:
It's not currently clear for me why is the task ever storred into a field, could you elaborate?
Can it live only on stack?

Reply:
I'm a bit confused by this comment. What's the alternative?